### PR TITLE
Add Release Instructions

### PR DESCRIPTION
**Pull request by tianskylan**
Moving "release instructions" outside of `Generator` directory, into its own document living under project root.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Run `$ make publish VERSION_NUMBER` at the root directory of needle (where the `Makefile` is located)
+
+For example:
+```
+$ make publish 0.13.0
+```
+
+This runs the steps specified in the `Makefile`.
+
+The `Makefile` does support failure recovering yet. If a certain step fails, you might want to figure out which step failed, and start manually run the remaining steps.
```
Comment (elsh) [inline RELEASE.md]: you mean 'does not support'? 

s/recovering/recovery 

remove 'start'

Comment (tianskylan) [reply RELEASE.md]: Nice, thanks for catching those errors. Fixed

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Run `$ make publish VERSION_NUMBER` at the root directory of needle (where the `Makefile` is located)
+
+For example:
+```
+$ make publish 0.13.0
+```
+
+This runs the steps specified in the `Makefile`.
+
+The `Makefile` does not support failure recovery yet. If a certain step fails, you might want to figure out which step failed, and manually run the remaining steps.
```
Comment (elsh) [inline RELEASE.md]: Add 'a' before failure. 
'you might want to' sounds casual, maybe better state 'it needs to be resolved and the remaining steps should be run manually.'

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+## Create a new Github release
+After all steps in the `Makefile` finish successfully, go to the Releases tab and create a new release.
```
Comment (elsh) [inline RELEASE.md]: all the steps

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+For example:
+```
+$ make publish 0.13.0
+```
+
+This runs the steps specified in the `Makefile`.
+
+The `Makefile` does not support failure recovery yet. If a certain step fails, you might want to figure out which step failed, and manually run the remaining steps.
+
+## Create a new Github release
```
Comment (elsh) [inline RELEASE.md]: Is this for external devs? Do we want to allow them to publish releases?  @rudro

Comment (tianskylan) [reply RELEASE.md]: I don't think we want to allow allow non-admins to make releases.
As written in the first paragraph of this doc, it's intended for collaborating admins like myself.

Comment (elsh) [reply RELEASE.md]: My point is, do we want to share this info on readme whose target audience is external devs?  I'll leave this up to @rudro.

Comment (rudro) [reply RELEASE.md]: I think it's fine to be public. It's relatively standard steps.  True it's only useful to the 2-3 people with admin access to this repo, but don't think it should confuse anyone.

**Pull request merged**